src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,343 @@
+package school.lemon.changerequest.java.io.hw1;
+
+
+import school.lemon.changerequest.java.io.hw1.exception.*;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+
+public class IFileManagerImp implements IFileManager {
+
+
+    File currentDirectory;
+
+    boolean directoriesFirst = true;
+
+
+    public IFileManagerImp(String path) {
+
+        this.currentDirectory = new File(path);
+    }
+
+    public IFileManagerImp() {
+        this.currentDirectory = new File(System.getProperty("user.dir"));
+    }
+
+    @Override
+    public String ls() {
+        return FilesInfo(filter(Order.BY_LEXICOGRAPHICAL_ASCENDING, true, fileFilterForAll()));
+    }
+
+    @Override
+    public String ls(Order order) {
+        return FilesInfo(filter(order, true, fileFilterForAll()));
+    }
+
+    @Override
+    public String ls(Order order, boolean directoriesFirst) {
+
+        return FilesInfo(filter(order, directoriesFirst, fileFilterForAll()));
```

Comment:
it's OK to move functionality into other method, but in my opinion - there is no need for other function. just move your implementation to `ls(order, directoriesFirst)` method from `FilesInfo` method.